enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,216 @@
+            course_start_date, "%Y-%m-%dT%H:%M:%SZ"
+        )
+        enrollment_unenrolled_at_datetime = datetime.strptime(
+            enrollment_unenrolled_at, "%Y-%m-%dT%H:%M:%SZ"
+        )
+        refund_cutoff_date = max(course_start_datetime, enrollment_created_datetime) + timedelta(days=14)
+
+        if refund_cutoff_date > enrollment_unenrolled_at_datetime:
+            logger.info(
+                f"Course run: {enrollment_course_run_key} is refundable for enterprise customer user: "
+                f"{enterprise_course_enrollment.get('enterprise_customer_user')}. Writing Reversal record."
+            )
+            return True
+        else:
+            logger.info(
+                f"Unenrollment from course: {enrollment_course_run_key} by user: "
+                f"{enterprise_course_enrollment.get('enterprise_customer_user')} is not refundable."
+            )
+            return False
+
+    def handle(self, *args, **options):
+        """
+        Fetch enterprise enrollment objects that have been unenrolled within the last 24 hours and write Transaction
+        Reversals where appropriate for each unenrollment.
+        """
+        DRY_RUN_LOG_PREFIX = ""
+        if dry_run := options.get('dry_run'):
+            DRY_RUN_LOG_PREFIX = "DRY RUN: "
+            logger.info("Running in dry-run mode. No updates or creates will occur.")
+
+        logger.info(
+            f"{DRY_RUN_LOG_PREFIX}Updating and writing Transaction Reversals from recent Enterprise unenrollments data"
+        )
+        recent_unenrollments = EnterpriseApiClient().fetch_recent_unenrollments()
+        logger.info(
+            f"{DRY_RUN_LOG_PREFIX}Found {len(recent_unenrollments)} recent Enterprise unenrollments"
+        )
+
+        reversals_processed = 0
+        fetched_content_metadata = {}
+        for unenrollment in recent_unenrollments:
+            fulfillment_uuid = unenrollment.get("uuid")
+            enterprise_course_enrollment = unenrollment.get("enterprise_course_enrollment")
+            enrollment_course_run_key = enterprise_course_enrollment.get("course_id")
+            enrollment_unenrolled_at = enterprise_course_enrollment.get("unenrolled_at")
+
+            # Look for a transaction related to the unenrollment
+            related_transaction = Transaction.objects.filter(
+                uuid=unenrollment.get('transaction_id')
+            ).first()
+            if not related_transaction:
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}No Subsidy Transaction found for enterprise fulfillment: {fulfillment_uuid}"
+                )
+                continue
+
+            idempotency_key = generate_transaction_reversal_idempotency_key(
+                fulfillment_uuid,
+                enrollment_unenrolled_at
+            )
+
+            # Look for a Reversal related to the unenrollment
+            existing_reversal = Reversal.objects.filter(
```

Comment:
you shouldn't filter by idempotency key here - as any valid reversal should be enough to skip this part - maybe `Transaction.get_reversal()` here?